src/test/java/com/cag/cagbackendapi/integration/ProfileControllerIntegrationTests.kt

```diff
@@ -186,4 +195,121 @@ class ProfileControllerIntegrationTests {
         assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.CONFLICT_MESSAGE)
         assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.USER_HAS_PROFILE)
     }
+
+    @Test
+    fun getProfile_validInput_200Success() {
+        //create user headers
+        val headers = HttpHeaders()
+        headers.set("authKey", validAuthKey)
+        val request = HttpEntity(validRegisterUser, headers)
+
+        //create user
+        val createdUserResponse = testRestTemplate.postForEntity("/user/register", request, String::class.java)
+        val createUser = objectMapper.readValue(createdUserResponse.body, UserDto::class.java)
+        val userIdUUID = createUser.userId
+
+        //register profile headers
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", validAuthKey)
+        headers2.set("userId", userIdUUID.toString())
```

Comment:
The UserId is sent in the path, not in the header! You also have it in the path so this test will pass, but this header is just extra unneeded data